assignments/hw1/ftregister.go

```diff
@@ -0,0 +1,114 @@
+package ftregister
+
+import (
+	"fmt"
+	"github.com/dati-mipt/distributed-storage-algorithms/network"
+	"github.com/dati-mipt/distributed-storage-algorithms/util"
+)
+
+type FaultTolerantRegister struct {
+	rid      int64
+	current  util.TimestampedValue
+	replicas map[int64]network.Link
+}
+
+func NewFaultTolerantRegister(rid int64) *FaultTolerantRegister {
+	return &FaultTolerantRegister{
+		rid:      rid,
+		replicas: map[int64]network.Link{},
+	}
+}
+
+func (r *FaultTolerantRegister) Write(value int64) bool {
+	fmt.Println("-----------Write-----------")
+	fmt.Printf("write value: %d\n", value)
+
+	r.current.Val = value
+	r.current.Ts = util.Timestamp{Number: r.current.Ts.Number + 1, Rid: r.rid}
```

Comment:
There is no interaction with the other replicas.
If this replica fall, all data will be vanished.